tests/v2/validation/provisioning/rke2/custom_cluster_test.go

```diff
@@ -96,82 +86,47 @@ func (c *CustomClusterProvisioningTestSuite) TestProvisioningRKE2CustomCluster()
+		{"1 Node all roles " + provisioning.AdminClientName.String(), c.client, nodeRolesAll, false},
+		{"1 Node all roles " + provisioning.StandardClientName.String(), c.standardUserClient, nodeRolesAll, false},
+		{"2 nodes - etcd/cp roles per 1 node " + provisioning.AdminClientName.String(), c.client, nodeRolesShared, false},
+		{"2 nodes - etcd/cp roles per 1 node " + provisioning.StandardClientName.String(), c.standardUserClient, nodeRolesShared, false},
+		{"3 nodes - 1 role per node " + provisioning.AdminClientName.String(), c.client, nodeRolesDedicated, false},
+		{"3 nodes - 1 role per node " + provisioning.StandardClientName.String(), c.standardUserClient, nodeRolesDedicated, false},
+		{"4 nodes - 1 role per node + 1 windows worker" + provisioning.AdminClientName.String(), c.client, nodeRolesDedicatedWindows, true},
+		{"4 nodes - 1 role per node + 1 windows worker" + provisioning.StandardClientName.String(), c.standardUserClient, nodeRolesDedicatedWindows, true},
 	}
-	var name string
 	for _, tt := range tests {
 		testSession := session.NewSession()
 		defer testSession.Cleanup()
-
-		client, err := tt.client.WithSession(testSession)
-		require.NoError(c.T(), err)
 		if (c.isWindows == tt.isWindows) || (c.isWindows && !tt.isWindows) {
-			for _, nodeProviderName := range c.nodeProviders {
-				externalNodeProvider := provisioning.ExternalNodeProviderSetup(nodeProviderName)
-				providerName := " Node Provider: " + nodeProviderName
-				for _, kubeVersion := range c.kubernetesVersions {
-					name = tt.name + providerName + " Kubernetes version: " + kubeVersion
-					for _, cni := range c.cnis {
-						name += " cni: " + cni
-						c.Run(name, func() {
-							TestProvisioningRKE2CustomCluster(c.T(), client, externalNodeProvider, tt.nodeRoles, tt.psact, kubeVersion, cni, c.hardened, c.advancedOptions)
-						})
-					}
-				}
-			}
+			permutations.RunTestPermutations(&c.Suite, tt.name, tt.client, c.clustersConfig, permutations.RKE2CustomCluster, nil, nil)
 		} else {
 			c.T().Skip("Skipping Windows tests")
 		}
 	}
 }
 
 func (c *CustomClusterProvisioningTestSuite) TestProvisioningRKE2CustomClusterDynamicInput() {
-	clustersConfig := new(provisioning.Config)
-	config.LoadConfig(provisioning.ConfigurationFileKey, clustersConfig)
-	nodesAndRoles := clustersConfig.NodesAndRoles
-	if len(nodesAndRoles) == 0 {
+	if len(c.clustersConfig.NodesAndRoles) == 0 {
 		c.T().Skip()
 	}
-
+	poolNames := permutations.GetTestName(c.clustersConfig.NodesAndRoles)
 	tests := []struct {
 		name   string
 		client *rancher.Client
-		psact  string
 	}{
-		{provisioning.AdminClientName.String(), c.client, c.psact},
-		{provisioning.StandardClientName.String(), c.standardUserClient, c.psact},
+		{provisioning.AdminClientName.String() + poolNames, c.client},
+		{provisioning.StandardClientName.String() + poolNames, c.standardUserClient},
```

Comment:
I still think we shouldn't add the pool names to the test name. I think it would be better to just log it.